src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -2071,4 +2071,35 @@ RubyArray zlibGetCRCTable() {
         }
     }
 
+    @CoreMethod(names = "rb_frame_method_and_id", onSingleton = true, required = 2)
+    public abstract static class FrameMethodAndId extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        boolean frameMethodAndId(Object frameMethod, Object frameId) {
+            final Frame callingMethodFrame = findCallingMethodFrame();
+            frameMethod = RubyArguments.getMethod(callingMethodFrame);
+            frameId = System.identityHashCode(RubyArguments.tryGetSelf(callingMethodFrame));
+            return true;
+        }
+
+        @TruffleBoundary
+        private static Frame findCallingMethodFrame() {
```

Comment:
I think org.truffleruby.language.CallStackManager#getCallingMethod should work instead of this and be a lot cleaner.